schedule/schedule-appengine/src/main/client/jasify/booking-via-jasify/booking-via-jasify.js

```diff
@@ -218,31 +218,93 @@
         }
 
         function bookIt() {
-            ShoppingCart.clearUserCart().then(function () {
-                var promises = [];
+            var request = {activityIds: [], activityPackageSubscriptions: []};
 
-                angular.forEach(vm.activitySelection, function (value) {
-                    $log.debug("Adding activity to shopping cart: " + value.id);
-                    promises.push(ShoppingCart.addUserActivity(value.id));
-                });
+            angular.forEach(vm.activitySelection, function (value) {
+                $log.debug("Adding activity to shopping cart: " + value.id);
+                this.activityIds.push(value.id);
+            }, request);
+
+            var completeActivityPackagesList = _.filter(vm.activityPackages, vm.packageSelectionComplete);
+
+            angular.forEach(completeActivityPackagesList, function (value) {
+                $log.debug("Adding activity package to shopping cart: " + value.id);
 
-                var completeActivityPackagesList = _.filter(vm.activityPackages, vm.packageSelectionComplete);
+                var activityIds = [];
+                angular.forEach(vm.activityPackageSelection[value.id], function (activity) {
+                    this.push(activity.id);
+                }, activityIds);
 
-                angular.forEach(completeActivityPackagesList, function (value) {
-                    $log.debug("Adding activity package to shopping cart: " + value.id);
-                    promises.push(ShoppingCart.addUserActivityPackage(value, vm.activityPackageSelection[value.id]));
+                this.activityPackageSubscriptions.push({
+                    activityPackageId: value.id,
+                    activityIds: activityIds
                 });
+            }, request);
+
+            ShoppingCart.createAnonymousCart(request).then(confirmPopup, error);
+
+            function error(r) {
+                jasDialogs.error(r.statusText + ' (' + r.status + ')');
+            }
+        }
 
-                $q.all(promises).then(function () {
-                    $log.debug('Shopping cart is ready');
-                    BrowserData.setPaymentAcceptRedirect('done');
-                    BrowserData.setPaymentCancelRedirect($location.path());
-                    $location.path('/checkout');
-                }, function () {
-                    // TODO
+        function confirmPopup(r) {
+            jasDialogs.ok('Proceed in new window', 'Booking is done via Jasify. ' +
+                'A new window will be opened where you will be able to Sign In or Create an account and ' +
+                'proceed to your checkout.',
+                onOk);
+            function onOk() {
+                var w = $window.innerWidth || 820;
```

Comment:
I popup a window based on the size of the current window `$window.innerWidth`.  The 820 is just a fallback.  This one I think is safe.